### ISSUE TITLE
Mempool Collector Stats

### ISSUE THREAD

**Issue by metachris** [flashbots/mempool-archiver#1]
Some early stats about transactions collected and stored with the mempool-collector:

### Data collection

Per hour:

- 70k - 100k transactions
- 350 - 500MB JSON files

Extrapolated to a day:
- Up to 2.5M transactions 
- Up to 12 GB raw storage

### Data size & compression

Looking at one particular hour specifically: 2023-08-04 UTC between [01:00, 02:00[:

* Unique tx: 78,757 (`find ./ -type f | wc -l`)
* Disk usage: 373 MB (`du --si -s`)
* Apparent size: 134 MB (`du --si -s --apparent-size`)
* Average file size: 1.584 KB (`ls -l | gawk '{sum += $5; n++;} END {print n" "sum" "sum/n;}'`)

`gzip` individual JSON files:

* Typical reduction: 50% 
* Interesting: `du` output of apparent-size vs disk usage:

```
$ du --si -s *
373M    h01
350M    h01_gz

$ du --si -s * --apparent-size
134M    h01
76M     h01_gz
```

zipping a whole daily folder:

```
$ zip -r h01 h01

$ ls -alh h01.zip
-rw-r--r-- 1 ubuntu ubuntu 74M Aug  4 10:30 h01.zip

$ du --si h01.zip
77M     h01.zip

$ du --si h01.zip --apparent-size
77M     h01.zip
```

**Comment by 0x416e746f6e** [comment]
indeed, disk usage reports the actually used disk space by the file.  since the filesystem stored data in blocks, then a non-sparse file will occupy `diskBlockSize * roundUp(fileSize / diskBlockSize) >= fileSize`.  therefore zipping many small (smaller than `diskBlockSize`) files is not going to change much with respect to actual disk usage.

**Comment by Miguelmavs** [comment]
$ du --si -s *
373M    h01
350M    h01_gz

$ du --si -s * --apparent-size
134M    h01
76M     h01_gz

**Comment by metachris** [comment]
Some stats after creating a summary file of 1,423,508 transactions in JSON and Parquet format:

| Format | Size |
|--------|--------|
| CSV | 314 MB |
| Parquet | 118 MB |

**Comment by metachris** [comment]
Perhaps the individual tx JSON file should also not contain the signature, to save 20-40% of the storage space, and because the signature is part of the rawTx anyway 🤔

**Comment by ra--** [comment]
everything is part of the rawtx, apart from timestamp and chainId

**Comment by metachris** [comment]
It's still convenient for the summarizer service not needing to parse every single rawTx and extracting the fields, although that doesn't seem too much to ask either.

I'm still undecided whether it's preferable to have the collector store [some fields](https://github.com/flashbots/mempool-archiver/blob/main/docs/example-tx-summary.json), or only store rawTx + timestamp

**Comment by metachris** [comment]
On a semi-related note: i'm still undecided whether we should store the timestamp in millis or microseconds. The problem being that microsecond precision might give the wrong impression -- that it's a reliable number. But imagine the data is coming over the network from a more distant EL node, even milliseconds will not be entirely reliable due to the network latency.
